node/router/messages/src/lib.rs

```diff
@@ -105,7 +162,40 @@ impl<N: Network> From<DisconnectReason> for Message<N> {
     }
 }
 
+impl<N: Network> TryFrom<Vec<MessageChunk>> for Message<N> {
+    type Error = io::Error;
+
+    fn try_from(chunks: Vec<MessageChunk>) -> io::Result<Self> {
+        if chunks.is_empty() {
+            return Err(error("A message can't be reconstructed from an empty list of chunks"));
+        }
+
+        // Recreate the full Message from the chunks.
+        let mut full = Vec::new();
+        for chunk in &chunks {
+            full.extend_from_slice(&chunk.blob);
+        }
+        let full_len = full.len();
+
+        // Verify the hash of the recreated message.
+        let expected_hash = chunks.first().unwrap().hash;
+        let recreated_hash = Sha256::digest(&full);
+        if *recreated_hash != expected_hash {
+            return Err(error("The hash of a chunked message doesn't match its expected value"));
+        }
+
+        // Parse the full Message.
+        let mut reader = io::Cursor::new(full);
+        let message = Message::read_le(&mut reader)?;
+        debug!("recreated a {}B {} from {} chunks", full_len, message.name(), chunks.len());
+
+        Ok(message)
+    }
+}
+
 impl<N: Network> Message<N> {
+    /// The maximum size of a chunk in a chunked message.
+    pub const MAX_CHUNK_LEN: usize = 64 * 1024;
```

Comment:
GPT question: Can you confirm if there's any clear **theoretical** performance benefit to chunking, if so how much roughly and how may it vary with chunk size?